test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPushCommandTest.cs

```diff
@@ -2078,6 +2084,339 @@ public void PushCommand_PushToServerV3_WithSymbols_ApiKey_SymbolApiKey_BothFromC
+                };
+
+                var result = CommandRunner.Run(
+                                NuGetExePath,
+                                testDirectory,
+                                string.Join(" ", args),
+                                true);
+
+                // Assert
+                Assert.True(
+                   result.ExitCode == 0,
+                   "Execution failed with:" + result.Output);
+
+                Assert.True(
+                    File.Exists(Path.Combine(repository2Path, packageFileName)),
+                    string.Format(
+                        CultureInfo.InvariantCulture,
+                        @"The package {0} was not found in {1}",
+                        packageFileName, repository2Path));
+            }
+        }
+
+        [Fact]
+        public void PushCommand_ConfigFile_NoSourceParam()
+        {
+            var configFileName = "My.Config";
+            var testRepository = "repository";
+
+            using (var testDirectory = TestDirectory.Create())
+            {
+                var repositoryPath = Path.Combine(testDirectory, testRepository);
+                var configFilePath = Path.Combine(testDirectory, "config");
+                Directory.CreateDirectory(repositoryPath);
+                Directory.CreateDirectory(configFilePath);
+
+                var packageFileName = Util.CreateTestPackage("testPackage1", "1.1.0", testDirectory);
+
+                FileSystemUtils.CreateFile(
+                    configFilePath,
+                    configFileName,
+                    string.Format(
+                        CultureInfo.InvariantCulture,
+                        @"<?xml version=""1.0"" encoding=""utf-8""?>
+                        <configuration>
+                            <config>
+                                <add key=""defaultPushSource"" value=""{0}"" />
+                            </config>
+                        </configuration>",
+                        repositoryPath));
+
+                var args = new string[]
+                {
+                    Command_Push,
+                    packageFileName,
+                    Param_ConfigFile,
+                    Path.Combine(configFilePath, configFileName)
+                };
+
+                var result = CommandRunner.Run(
+                                NuGetExePath,
+                                testDirectory,
+                                string.Join(" ", args),
+                                true);
+
+                Assert.True(
+                   result.ExitCode == 0,
+                   "Execution failed with:" + result.Output);
+
+                Assert.True(
+                    File.Exists(Path.Combine(repositoryPath, packageFileName)),
+                    string.Format(
+                        CultureInfo.InvariantCulture,
+                        @"The package {0} was not found in {1}",
+                        packageFileName, repositoryPath));
+            }
+        }
+
```

Comment:
Is there any happy path test added in this file? Or it already exists?

Reply:
I added 7 tests to this file. The one covering the "happy path" is PushCommand_ConfigFile_ExpectedFormat.